libs/barista-components/formatters/src/duration/duration-formatter-constants.ts

```diff
@@ -33,19 +33,19 @@ export function toDurationMode(
 // tslint:disable: no-magic-numbers
 /** Factorials needed for converting milliseconds to other time units */
 export const CONVERSION_FACTORS_TO_MS = new Map<DtTimeUnit, number>([
-  [DtTimeUnit.YEAR, 12 * 30.41666 * 24 * 60 * 60 * 1000],
-  [DtTimeUnit.MONTH, 30.41666 * 24 * 60 * 60 * 1000],
-  [DtTimeUnit.DAY, 24 * 60 * 60 * 1000],
-  [DtTimeUnit.HOUR, 60 * 60 * 1000],
-  [DtTimeUnit.MINUTE, 60 * 1000],
-  [DtTimeUnit.SECOND, 1000],
-  [DtTimeUnit.MILLISECOND, 1],
-  [DtTimeUnit.MICROSECOND, 1000], // Has to be handled differently because IEEE can't handle floating point numbers very well
-  [DtTimeUnit.NANOSECOND, 1], // Has to be handled differently because IEEE can't handle floating point numbers very well
+  [DtTimeUnit.YEAR, 12 * 30.41666 * 24 * 60 * 60 * 1000000000],
```

Comment:
These are really hard to read but there is a cool trick you can use to make it easier. 
```suggestion
  [DtTimeUnit.YEAR, 12 * 30.41666 * 24 * 60 * 60 * 1_000_000_000],
```

Reply:
I'd also suggest calculate them step by step. 
start with nano as 1.
Next use nano and multiply by 1000. Next use microsecond and multiply by 1000. This makes it easier to see where you are at in the chain. Store them in the map at the end.